dask/array/rechunk.py

```diff
@@ -7,26 +7,31 @@
 """
 from __future__ import absolute_import, division, print_function
 
+import heapq
+
 from itertools import product, chain
-from operator import getitem, add
+from operator import getitem, add, mul, itemgetter
 
 import numpy as np
-from toolz import merge, accumulate
+from toolz import merge, accumulate, reduce
 
 from ..base import tokenize
 from .core import concatenate3, Array, normalize_chunks
 
 
+def cumdims_label_1d(blocks, const):
```

Comment:
Right, I originally called this in the previous version of `plan_rechunk()`, but I can fold it again.